File: samples/cdk/python/quickstart_rotation.py
"""Quickstart: Create a ThresholdRotationPolicy and test deactivation logic.

Demonstrates how the threshold-based rotation policy decides whether a
model should be deactivated based on its failure count and error rate.
"""

from modelmesh.cdk import ThresholdRotationPolicy, ThresholdRotationConfig
from modelmesh.cdk.helpers import mock_model_snapshot


def main() -> None:
    policy = ThresholdRotationPolicy(ThresholdRotationConfig(
        failure_count_threshold=5,
        error_rate_threshold=0.3,
        cooldown_seconds=120,
    ))

    healthy = mock_model_snapshot(failure_count=0)
    print(f"Healthy model deactivate? {policy.should_deactivate(healthy)}")  # False

    failing = mock_model_snapshot(failure_count=6)
    print(f"Failing model deactivate? {policy.should_deactivate(failing)}")  # True
    print(f"Reason: {policy.get_reason(failing)}")  # ERROR_THRESHOLD


if __name__ == "__main__":
    main()

File: samples/cdk/typescript/quickstartRotation.ts
/**
 * Quickstart: Create a BaseRotationPolicy and test deactivation logic.
 *
 * Demonstrates how the threshold-based rotation policy decides whether a
 * model should be deactivated based on its failure count and error rate.
 */

import {
    BaseRotationPolicy,
    BaseRotationConfig,
    ModelState,
    ModelStatus,
    DeactivationReason,
    createDefaultModelState,
} from "@nistrapa/modelmesh-core";

function main(): void {
    const policy = new BaseRotationPolicy({
        failureThreshold: 5,
        errorRateThreshold: 0.3,
        cooldownSeconds: 120,
    });

    const healthy = createDefaultModelState({ modelId: "gpt-4o", failureCount: 0 });
    console.log(`Healthy model deactivate? ${policy.shouldDeactivate(healthy)}`);  // false

    const failing = createDefaultModelState({ modelId: "gpt-4o", failureCount: 6 });
    console.log(`Failing model deactivate? ${policy.shouldDeactivate(failing)}`);  // true
    console.log(`Reason: ${policy.getReason(failing)}`);  // ERROR_THRESHOLD
}

main();
